python/python/tests/test_fragment.py

```diff
@@ -459,3 +459,15 @@ def test_fragment_metadata_pickle(tmp_path: Path, enable_move_stable_row_ids: bo
     round_trip = pickle.loads(pickle.dumps(frag_meta))
 
     assert frag_meta == round_trip
+
+
+def test_write_fragment_with_file_writer_option(tmp_path: Path):
+    df = pd.DataFrame({"a": [1, 2, 3, 4, 5]})
+    file_writer_options = {"data_cache_bytes": "4096", "max_page_bytes": "4096"}
+    frag = LanceFragment.create(tmp_path, df, file_writer_options=file_writer_options)
+
+    assert len(frag.files) == 1
+    assert frag.files[0].fields == [0]
+    assert frag.physical_rows == 5
+    assert frag.row_id_meta is None
+    assert frag.deletion_file is None
```

Comment:
These tests don't actually validate the page size is being set correctly but I can't think of an easy way to do so :shrug: 

Maybe we could open the written data file directly with `LanceFileReader` and look at the metadata?

Reply:
I added a test case in the dataset test suite using `len(reader.metadata().columns.pages)`. 

But I did check the buffer size of pages. It seems the size is 2736 not 4096 set in options.
![image](https://github.com/user-attachments/assets/6cb533f4-0afc-4918-a146-5a435d59b9bc)

Reply:
I'm not sure all the buffer size is 2736, so I do not assert the size.